cli/azd/pkg/project/service_target_dotnet_containerapp.go

```diff
@@ -455,6 +450,28 @@ func (fns *containerAppTemplateManifestFuncs) ConnectionString(name string) (str
 	case "azure.sql.v0", "sqlserver.server.v0":
 		return fns.sqlConnectionString(name, "")
 	case "azure.sql.database.v0", "sqlserver.database.v0":
+		parentResource := resource.Parent
+		if parentResource == nil || *parentResource == "" {
+			return "", fmt.Errorf("parent resource not found for db: %s", name)
+		}
+
+		parent := fns.manifest.Resources[*parentResource]
+		if parent.Type == "container.v0" {
+			rawConnectionString := strings.Split(strings.TrimRight(*parent.ConnectionString, ";"), ";")
+			rawConnectionString = append(rawConnectionString, fmt.Sprintf("Database=%s;", name))
+			for i, part := range rawConnectionString {
+				keyValue := strings.Split(part, "=")
```

Comment:
Is it better to provide a more decent error message the `key=value` if it's malformed (`len(keyValue)< 2`)? Based on [connection strings docs](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/connection-strings), it's guaranteed to be a key-value tuple

Reply:
Based on the description of the format, I think `SplitN(part, "=", 1)` is more correct. There can be unescaped `=` characters within the value.

Another thought here is to avoid connection string parsing, and strictly lex for bracketed `{}` expressions and evaluate those.

Reply:
Now that you mention it, `EvalString()` is already finding all the `{ fooExp }` and running the `func(exp string) string`,  so we can apply it directly to the full connectionString :)
Thank you!